pkg/providers/instance/instance.go

```diff
@@ -289,6 +296,9 @@ func (p *DefaultProvider) getLaunchTemplateConfigs(ctx context.Context, nodeClas
 	var launchTemplateConfigs []*ec2.FleetLaunchTemplateConfigRequest
 	launchTemplates, err := p.launchTemplateProvider.EnsureAll(ctx, nodeClass, nodeClaim, instanceTypes, capacityType, tags)
 	if err != nil {
+		if cloudprovider.IsInsufficientCapacityError(err) {
```

Comment:
Same comment here about error wrapping and checking. You shouldn't need to do this as long as the error that's returned from L297 is an ICE error